src/replication.c

```diff
@@ -323,26 +467,24 @@ void replicationFeedSlaves(list *slaves, int dictid, robj **argv, int argc) {
  * guess what kind of bug it could be. */
 void showLatestBacklog(void) {
     if (server.repl_backlog == NULL) return;
+    if (listLength(server.repl_buffer_blocks) == 0) return;
```

Comment:
Seems to me that our assumption is the following, so we should assert for it:
```suggestion
    if (server.repl_backlog == NULL) {
        serverAssert(listLength(server.repl_buffer_blocks) == 0);
        return;
    }
```

Reply:
This comment is resolved without a response or code change.